api/server/sdk/server_interceptors.go

```diff
@@ -0,0 +1,198 @@
+/*
+Package sdk is the gRPC implementation of the SDK gRPC server
+Copyright 2018 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sdk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	sdk_auth "github.com/libopenstorage/openstorage-sdk-auth/pkg/auth"
+	"github.com/pborman/uuid"
+	"github.com/sirupsen/logrus"
+
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Keys to store data in gRPC context. Use these keys to retrieve
+// the data from the gRPC context
+type InterceptorContextkey string
+
+const (
+	// Key to store in the token claims in gRPC context
+	InterceptorContextTokenKey InterceptorContextkey = "tokenclaims"
+)
+
+var (
+	defaultRoles = map[string][]sdk_auth.Rule{
+		"admin": {
+			{
+				Services: []string{"*"},
+				Apis:     []string{"*"},
+			},
+		},
+	}
+)
+
+// This interceptor provides a way to lock out any calls while we adjust the server
+func (s *sdkGrpcServer) rwlockIntercepter(
+	ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+) (interface{}, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	return handler(ctx, req)
+}
+
+// Authenticate user and add authorization information back in the context
+func (s *sdkGrpcServer) auth(ctx context.Context) (context.Context, error) {
+	token, err := grpc_auth.AuthFromMD(ctx, "bearer")
```

Comment:
nit: bearer could be a constant, with some comment explaining what it is and why we are not using `basic`

Reply:
Will do. I will add that it is just standard JWT style